src/proxy/outbound.rs

```diff
@@ -180,72 +183,98 @@ impl OutboundConnection {
     }
 
     async fn build_request(&self, downstream: IpAddr, target: SocketAddr) -> Request {
-        let (source_workload, us, is_vip) = {
-            let source_workload = self
-                .workloads
-                .fetch_workload(&downstream)
-                .await
-                .expect("todo: source must be found");
+        let source_workload = self
+            .workloads
+            .fetch_workload(&downstream)
+            .await
+            .expect("todo: source must be found");
 
-            // TODO: we want a single lock for source and upstream probably...?
-            let (us, is_vip) = self.workloads.find_upstream(target).await;
-            (source_workload, us, is_vip)
-        };
-        let mut req = Request {
-            protocol: us.workload.protocol,
-            source: source_workload.clone(), // TODO drop clone
-            destination: SocketAddr::from((us.workload.workload_ip, us.port)),
-            gateway: us
-                .workload
-                .gateway_ip
-                .expect("todo: refactor gateway ip handling"),
-            direction: Direction::Outbound, // TODO set this
-            request_type: RequestType::Direct,
-        };
-        if source_workload.waypoint_address.is_some() {
-            // Source has a remote proxy. We should delegate everything to that proxy - do not even resolve VIP.
-            // TODO: add client skipping
-            req.request_type = RequestType::ToClientWaypoint;
-            // Let the client remote know we are on the outbound path. The remote proxy should strictly
-            // validate the identity when we declare this
-            req.direction = Direction::Outbound;
-            // Load balancing decision is deferred to remote proxy
-            req.destination = target;
-            // Send to the remote proxy
-            req.gateway = SocketAddr::from((source_workload.waypoint_address.unwrap(), 15001));
-            // Always use HBONE here
-            req.protocol = Protocol::Hbone;
-        } else if us.workload.waypoint_address.is_some() {
-            // TODO: even in this case, we are picking a single upstream pod and deciding if it has a remote proxy.
-            // Typically this is all or nothing, but if not we should probably send to remote proxy if *any* upstream has one.
-            if is_vip {
```

Comment:
Yes, intentionally removed it since we only use it in the case for to upstream remote proxy(waypoint), no matter the original target is endpoint or vip, we could just use the original target.

Reply:
Why is the case of "upstream remote proxy(waypoint)" not important? or are you saying its the same anyways? Ill take a closer look

Reply:
Yes, I'm saying it's the same anyways. What we did for the case of "upstream remote proxy(waypoint)" is to set the "req.destination" to the original VIP if possible, so if the target is SVC IP, we set it to original target, if the target is directly POD endpoint, we also set it the to original target.  These two case could be combined no matter "is_vip".